### ISSUE TITLE
Attribute for either URL or unique identifier for obtaining the video

### ISSUE THREAD

**Issue by kyletpugh** [twrecked/hass-aarlo#454]
There is currently an attribute that is updated with the URL of the thumbnail image for video every recording. I use this to generate a notification with the image, I would like to have the notification either open to live streaming or to the video that generated the thumbnail depending on the choice from the actionable notification. Would it be possible to add another attribute that would provide the access to a specific video? As is I only seem to be able to access only the last video. I assume this is possible since your arlo-card works pretty well. Is it possible to open that card directly to a a specific video?

**Comment by twrecked** [comment]
There is already `last_video` attribute which points to the last recording? A new recording won't invalidate this URL because these links are directly to the videos on the Arlo servers.

The Lovelace cards work by calling the `aarlo_library` webservice which provides the last `x` number of videos.

I guess the question is, how are you reading the attributes at the moment?

**Comment by kyletpugh** [comment]
Thanks for your response. First, on what entity is the last_video attribute? Or do you mean the service call? Either way there's an issue where the last video is not always the same as the thumbnail because they are not updated at exactly the same time. Or with a unique identifier that all the attributes have been updated. If you have a trigger on the last activity entity without an attribute then it triggers multiple times. Basically, I'm sending a push notification that has choices and I want to either open to the video or to the live camera feed. However, I'm discovering that not only does this appear to be really hard but the first step in just getting the notification built it's seeming to not be possible currently. Also, I only see a lower 0.7 version but I see the repo is at 0.8. Do I have another issue that is messing this up?

**Comment by twrecked** [comment]
I think I understand what you're asking:

You'd like a `video_url` attribute, similar to the `thumbnail_url` attribute, added to the `sensor.aarlo_last_*` sensors?

And `0.7` is the latest,  `0.8` is an alpha release I'm working on.

**Comment by kyletpugh** [comment]
Yeah, that's what I was trying to say.... lol. Because I want the video and thumbnail url at the same time when they are both for the same video. Currently I'm getting a mix of the previous and next video, and the previous and next thumbnail. Sometimes the thumbnail is empty, I suspect this might be part of the other issue where it gets set to empty for a moment or maybe the Arlo thumbnail url for that image actually changed. I haven't been able to catch the trace for all of them... The versioning makes sense after I saw the other issue and adding it to 0.8. Thanks so much.

**Comment by twrecked** [comment]
I think that is relatively easy. I'll take a look tonight (it's morning here for reference!).

**Comment by twrecked** [comment]
I added the `video_url` in 0.7.1.1.

Turns out the other fix - don't update unless changed - is more complicated because the value does change. I'll think about that one some more.

**Comment by kyletpugh** [comment]
The url for the thumbnail changes? Like I'm assuming whenever the connection needs reauthenticated then I get new notifications but they are the same as the previous. There's no manual action in taking, and I only have the notification automation. Randomly I just get a notification for every camera. It happened at midnight last night and at 8am this morning. One of the cameras didn't even record anything for almost two days...

**Comment by kyletpugh** [comment]
I realize now that the video actually was not available at the time of the thumbnail. So I still have the same issue, either trigger on the thumbnail_url or the video_url. They don't match because the thumbnail is created immediately upon motion and the video whenever it's available after recording. The thumbnail can change again before the video does, lol. Need to be able to get something like a unique identifier that can get me the correct thumbnail and video when the motion starts so I can choose to watch it live or watch the recording (or nothing). I don't know...
